exporter/exporterhelper/internal/bounded_memory_queue_test.go

```diff
@@ -146,60 +146,79 @@ func TestShutdownWhileNotEmpty(t *testing.T) {
 }
 
 func Benchmark_QueueUsage_10000_1_50000(b *testing.B) {
-	queueUsage(b, 10000, 1, 50000)
+	benchmarkQueueUsage(b, 10000, 1, 50000)
 }
 
 func Benchmark_QueueUsage_10000_2_50000(b *testing.B) {
-	queueUsage(b, 10000, 2, 50000)
+	benchmarkQueueUsage(b, 10000, 2, 50000)
 }
 func Benchmark_QueueUsage_10000_5_50000(b *testing.B) {
-	queueUsage(b, 10000, 5, 50000)
+	benchmarkQueueUsage(b, 10000, 5, 50000)
 }
 func Benchmark_QueueUsage_10000_10_50000(b *testing.B) {
-	queueUsage(b, 10000, 10, 50000)
+	benchmarkQueueUsage(b, 10000, 10, 50000)
 }
 
 func Benchmark_QueueUsage_50000_1_50000(b *testing.B) {
-	queueUsage(b, 50000, 1, 50000)
+	benchmarkQueueUsage(b, 50000, 1, 50000)
 }
 
 func Benchmark_QueueUsage_50000_2_50000(b *testing.B) {
-	queueUsage(b, 50000, 2, 50000)
+	benchmarkQueueUsage(b, 50000, 2, 50000)
 }
 func Benchmark_QueueUsage_50000_5_50000(b *testing.B) {
-	queueUsage(b, 50000, 5, 50000)
+	benchmarkQueueUsage(b, 50000, 5, 50000)
 }
 func Benchmark_QueueUsage_50000_10_50000(b *testing.B) {
-	queueUsage(b, 50000, 10, 50000)
+	benchmarkQueueUsage(b, 50000, 10, 50000)
 }
 
 func Benchmark_QueueUsage_10000_1_250000(b *testing.B) {
-	queueUsage(b, 10000, 1, 250000)
+	benchmarkQueueUsage(b, 10000, 1, 250000)
 }
 
 func Benchmark_QueueUsage_10000_2_250000(b *testing.B) {
-	queueUsage(b, 10000, 2, 250000)
+	benchmarkQueueUsage(b, 10000, 2, 250000)
 }
 func Benchmark_QueueUsage_10000_5_250000(b *testing.B) {
-	queueUsage(b, 10000, 5, 250000)
+	benchmarkQueueUsage(b, 10000, 5, 250000)
 }
 func Benchmark_QueueUsage_10000_10_250000(b *testing.B) {
-	queueUsage(b, 10000, 10, 250000)
+	benchmarkQueueUsage(b, 10000, 10, 250000)
 }
 
-func queueUsage(b *testing.B, capacity int, numConsumers int, numberOfItems int) {
+func TestQueueUsage(t *testing.T) {
+	t.Run("with enough workers", func(t *testing.T) {
+		queueUsage(t, 10000, 5, 1000)
+	})
+	t.Run("past capacity", func(t *testing.T) {
+		queueUsage(t, 10000, 2, 50000)
+	})
+}
+
+func benchmarkQueueUsage(b *testing.B, capacity int, numConsumers int, numberOfItems int) {
 	b.ReportAllocs()
 	for i := 0; i < b.N; i++ {
-		q := NewBoundedMemoryQueue(capacity, numConsumers)
-		err := q.Start(context.Background(), componenttest.NewNopHost(), newNopQueueSettings(func(item Request) {
-			time.Sleep(1 * time.Millisecond)
-		}))
-		require.NoError(b, err)
-		for j := 0; j < numberOfItems; j++ {
-			q.Produce(newStringRequest(fmt.Sprintf("%d", j)))
+		queueUsage(b, capacity, numConsumers, numberOfItems)
+	}
+}
+
+func queueUsage(tb testing.TB, capacity int, numConsumers int, numberOfItems int) {
+
+	q := NewBoundedMemoryQueue(capacity, numConsumers)
+	var wg sync.WaitGroup
+	err := q.Start(context.Background(), componenttest.NewNopHost(), newNopQueueSettings(func(item Request) {
+		wg.Done()
+	}))
+	wg.Add(numberOfItems)
+	require.NoError(tb, err)
+	for j := 0; j < numberOfItems; j++ {
+		if !q.Produce(newStringRequest(fmt.Sprintf("%d", j))) {
```

Comment:
Why do we need to benchmark queue overflow rejects? Shouldn't we have assert here instead?

Reply:
We can have an assert too. The waitGroup waits at the end for completion of all items, or for all the rejections to be processed.

Reply:
I still don't get the reason of running benchmarks for queue overflow rejects. Can you please replace this with
```
assert.True(t, q.Produce(newStringRequest(fmt.Sprintf("%d", j)))
```
and update/remove the callers causing this assert to fail

Reply:
I'll just remove benchmarks causing overflows then, since they add no practical value.